barretenberg/cpp/src/barretenberg/circuit_checker/translator_circuit_checker.cpp

```diff
@@ -85,389 +85,408 @@ bool TranslatorCircuitChecker::check(const Builder& circuit)
-    // consider making use of relations.
-
-    for (size_t i = 2; i < circuit.num_gates - 1; i += 2) {
-        {
-            // Get the values of P.x
-            Fr op_code = circuit.get_variable(op_wire[i]);
-            Fr p_x_lo = circuit.get_variable(x_lo_y_hi_wire[i]);
-            Fr p_x_hi = circuit.get_variable(x_hi_z_1_wire[i]);
-            Fr p_x_0 = circuit.get_variable(p_x_0_p_x_1_wire[i]);
-            Fr p_x_1 = circuit.get_variable(p_x_0_p_x_1_wire[i + 1]);
-            Fr p_x_2 = circuit.get_variable(p_x_2_p_x_3_wire[i]);
-            Fr p_x_3 = circuit.get_variable(p_x_2_p_x_3_wire[i + 1]);
-            const std::vector p_x_binary_limbs = { p_x_0, p_x_1, p_x_2, p_x_3 };
-
-            // P.y
-            Fr p_y_lo = circuit.get_variable(y_lo_z_2_wire[i]);
-            Fr p_y_hi = circuit.get_variable(x_lo_y_hi_wire[i + 1]);
-            Fr p_y_0 = circuit.get_variable(p_y_0_p_y_1_wire[i]);
-            Fr p_y_1 = circuit.get_variable(p_y_0_p_y_1_wire[i + 1]);
-            Fr p_y_2 = circuit.get_variable(p_y_2_p_y_3_wire[i]);
-            Fr p_y_3 = circuit.get_variable(p_y_2_p_y_3_wire[i + 1]);
-            const std::vector p_y_binary_limbs = { p_y_0, p_y_1, p_y_2, p_y_3 };
-            // z1, z2
-            Fr z_1 = circuit.get_variable(x_hi_z_1_wire[i + 1]);
-            Fr z_2 = circuit.get_variable(y_lo_z_2_wire[i + 1]);
-
-            Fr z_1_lo = circuit.get_variable(z_lo_wire[i]);
-            Fr z_2_lo = circuit.get_variable(z_lo_wire[i + 1]);
-            Fr z_1_hi = circuit.get_variable(z_hi_wire[i]);
-            Fr z_2_hi = circuit.get_variable(z_hi_wire[i + 1]);
-
-            const std::vector z_1_binary_limbs = { z_1_lo, z_1_hi };
-            const std::vector z_2_binary_limbs = { z_2_lo, z_2_hi };
-            // Relation limbs
-            Fr low_wide_relation_limb = circuit.get_variable(relation_wide_limbs_wire[i]);
-            Fr high_wide_relation_limb = circuit.get_variable(relation_wide_limbs_wire[i + 1]);
-
-            // Current accumulator (updated value)
-            const std::vector current_accumulator_binary_limbs = {
-                circuit.get_variable(accumulators_binary_limbs_0_wire[i]),
-                circuit.get_variable(accumulators_binary_limbs_1_wire[i]),
-                circuit.get_variable(accumulators_binary_limbs_2_wire[i]),
-                circuit.get_variable(accumulators_binary_limbs_3_wire[i]),
-            };
-
-            // Previous accumulator
-            const std::vector previous_accumulator_binary_limbs = {
-                circuit.get_variable(accumulators_binary_limbs_0_wire[i + 1]),
-                circuit.get_variable(accumulators_binary_limbs_1_wire[i + 1]),
-                circuit.get_variable(accumulators_binary_limbs_2_wire[i + 1]),
-                circuit.get_variable(accumulators_binary_limbs_3_wire[i + 1]),
-            };
+    auto check_binary_limbs_equality = [&](const std::vector<Fr>& first, const std::vector<Fr>& second, size_t gate) {
```

Comment:
This is a large diff - is this just rearranging? is it related to the relation changes?